clients/drcachesim/tracer/tracer.cpp

```diff
@@ -1398,6 +1409,100 @@ event_delay_app_instruction(void *drcontext, void *tag, instrlist_t *bb, instr_t
     return DR_EMIT_DEFAULT;
 }
 
+
```

Comment:
How about putting this new function wrapping code in a separate file?  The C style here would be a separate file with an init and exit routine.  Looks like it needs to share per_thread_t and BUF_PTR -- though I would say just expose an append_marker routine and that can be the start of a future public interface.

Reply:
Done